docs/file-types/jupytext.md

```diff
@@ -0,0 +1,99 @@
+(file-types:custom)=
+# Custom notebook formats and Jupytext
+
+You can designate additional file types to be converted to Notebooks, and then executed / parsed in the same manner as regular Notebooks.
+
+```yaml
+sphinx:
+  config:
+    nb_custom_formats:
+        .mysuffix: mylibrary.converter_function
+```
+
+- The string should be a Python function that will be loaded by `import mylibrary.converter_function`
+- The function should take a file's contents (as a `str`) and return an [nbformat.NotebookNode](nbformat:api)
+
+If the function takes additional keyword arguments, then you can specify these as dictionary in a second argument.
+For example this is what the default conversion would look like:
+
+```yaml
+sphinx:
+  config:
+    nb_custom_formats:
+        .ipynb:
+            - nbformat.reads
+            - as_version: 4
+```
+
+:::{important}
+
+By default, Markdown cells in the Notebook will be parsed using the same MyST parser configuration as for other Markdown files.
+
+But, if this is incompatible with your file format, then you can specify for the Markdown to be parsed as **strictly CommonMark**, using a third argument:
+
+```yaml
+sphinx:
+  config:
+    nb_custom_formats:
+        .ipynb:
+            - nbformat.reads
+            - as_version: 4
+            - true
```

Comment:
I wonder if we should make this a key:val argument just to make it easier for users to know what they're configuring. e.g. `commonmark: true` rather than just `true`. WDYT?

Reply:
Yeh its currently adapted directly from https://nbsphinx.readthedocs.io/en/0.7.1/custom-formats.html.
Does it sound reasonable to add `custom_formats` to the `parse` key, and do the transform in `yaml_to_sphinx`, rather than changing it in `myst-nb`?

Reply:
??